docs/book/content/advanced/subscriptions.md

```diff
@@ -0,0 +1,165 @@
+# Subscriptions
+### How to achieve realtime data with GraphQL subscriptions
+
+GraphQL Subscriptions are a way to push data from the server to the clients that choose to listen to real time messages 
+from the server. Subscriptions are similar to queries in that they specify a set of fields to be delivered to the client,
+but instead of immediately returning a single answer, a result is sent every time a particular event happens on the 
+server. 
+
+For the usage of subscriptions you need to add [juniper_subscriptions][juniper_subscriptions] into your cargo.toml:
+```toml
+[dependencies]
+juniper = { git = "https://github.com/graphql-rust/juniper", branch = "master" }
+juniper_subscriptions = { git = "https://github.com/graphql-rust/juniper", branch = "master" }
+```
+
+### Schema Definition
+
+The Subscription is just a GraphQL object, similar to the Query root and Mutations object that you defined for the 
+operations in your [Schema][Schema], the difference is that all the operations defined there should be async and the return of it
+should be a [Stream][Stream].
+
+This example shows a subscription operation that returns two events, the strings `Hello` and `World!`
+sequentially: 
+
+```rust
+# use juniper::http::GraphQLRequest;
+# use juniper::{DefaultScalarValue, FieldError, SubscriptionCoordinator};
+# use juniper_subscriptions::Coordinator;
+# use futures::{Stream, StreamExt};
+# use std::pin::Pin;
+# #[derive(Clone)]
+# pub struct Database;
+# impl juniper::Context for Database {}
+# impl Database {
+#    fn new() -> Self {
+#        Self {}
+#    }
+# }
+# pub struct Query;
+# #[juniper::graphql_object(Context = Database)]
+# impl Query {
+#    fn hello_world() -> &str {
+#        "Hello World!"
+#    }
+# }
+pub struct Subscription;
+
+type StringStream = Pin<Box<dyn Stream<Item = Result<String, FieldError>> + Send>>;
+
+#[juniper::graphql_subscription(Context = Database)]
+impl Subscription {
+    async fn hello_world() -> StringStream {
+        let stream = tokio::stream::iter(vec![
+            Ok(String::from("Hello")),
+            Ok(String::from("World!"))
+        ]);
+        Box::pin(stream)
+    }
+}
+# fn main () {}
+```         
+
+
+
+### Coordinator
+
```

Comment:
I'd explain here a bit more what subscription coodinator is and why we need it. I copied a bit from [this comment](https://github.com/graphql-rust/juniper/pull/433#issuecomment-553570544)

```
Subscriptions require a bit more resources than regular queries, since they can provide 
a great vector for DOS attacks and can bring down a server easily if not handled right. 
[SubscriptionCoordinator][SubscriptionCoordinator] trait provides the coordination logic. 
It contains the schema and can keep track of opened connections, handle subscription 
start and maintains a global subscription id. Once connection is established, subscription 
coordinator spawns a [SubscriptionConnection][SubscriptionConnection], which handles a 
single connection, provides resolver logic for a client stream and can provide re-connection 
and shutdown logic.
```